sql/core/src/test/resources/tpcds-plan-stability/approved-plans-v1_4/q70.sf100/explain.txt

```diff
@@ -157,121 +158,125 @@ Input [2]: [s_state#14, sum#16]
 Keys [1]: [s_state#14]
 Functions [1]: [sum(UnscaledValue(ss_net_profit#10))]
 Aggregate Attributes [1]: [sum(UnscaledValue(ss_net_profit#10))#17]
-Results [3]: [s_state#14, s_state#14, MakeDecimal(sum(UnscaledValue(ss_net_profit#10))#17,17,2) AS _w2#18]
+Results [3]: [s_state#14 AS s_state#18, s_state#14, MakeDecimal(sum(UnscaledValue(ss_net_profit#10))#17,17,2) AS _w2#19]
 
-(25) Sort [codegen id : 5]
-Input [3]: [s_state#14, s_state#14, _w2#18]
-Arguments: [s_state#14 ASC NULLS FIRST, _w2#18 DESC NULLS LAST], false, 0
+(25) Exchange
```

Comment:
so we have an extra exchange now?

Reply:
Yes we have.
I think that's because in a `HashAggregate` in q70 we have an output attribute and we also have it aliased (we kept that alias with this PR to avoid the same output attribute appear multiple times).
IMO that shouldn't be a bad thing and the plan looks ok. But the root cause of the issue is in `AliasAwareOutputPartitioning` as it generates `outputPartitioning` based on the aliased output attribute only and doesn't produce a `PartitioningCollection` with both the base attribute and the aliased version.
As the parent of the aggregate requires a distribution based on the base attribute `EnsureRequirements` inserts the extra exchange...

We also have test failures, I'm not sure yet why.

Reply:
I put the WIP flag into the PR description for now and giving a try to an experiment with https://github.com/apache/spark/pull/37334/commits/b3b7d24b47c9748e6cff9211f6576048ba07cc8c to fix `AliasAwareOutputPartitioning` and `AliasAwareOutputOrdering` to handle cases when `AliasAwareOutputExpression.outputExpressions` contains both an attribute and its aliased version.

Reply:
https://github.com/apache/spark/commit/b3b7d24b47c9748e6cff9211f6576048ba07cc8c seems to fixed the issue. I fixed the indentation of the code and added some comments in https://github.com/apache/spark/pull/37334/commits/a7370203c875ea538628bd6127615aa8298b7cd0.

Very likely https://github.com/apache/spark/pull/37334/commits/59dcc0dfcabeec6f0998ddc2ee1299802b659cb8 will fix the other test issues, but tests are still running...